labgrid/driver/sigrokdriver.py

```diff
@@ -329,3 +330,94 @@ def measure(self):
+    works with other single chnnel DMMs.
+
+    This driver binds to a SigrokUsbDevice.
+    Make sure to select the correct driver for your DMM there.
+
+    Example usage:
+    > resources:
+    >   - SigrokUSBDevice:
+    >       driver: uni-t-ut61c
+    >       match:
+    >         'ID_PATH': pci-0000:07:00.4-usb-0:2:1.0
+    > drivers:
+    >   - SigrokDmmDriver: {}
+
+    Args:
+        bindings (dict): driver to use with sigrok
+    """
+    bindings = {
+        "sigrok": {SigrokUSBSerialDevice, NetworkSigrokUSBSerialDevice, SigrokUSBDevice, NetworkSigrokUSBDevice},
+    }
+
+    @Driver.check_active
+    @step(result=True)
+    def start(self, samples, timeout=None):
+        """
+        Starts to read samples from the DMM.
+        This function returns once sampling has been started. Sampling continues in the background.
+
+        Note: We use subprocess.PIPE to buffer the samples. When this buffer is too small for the number of
+        sampled requested sampling may stall.
+
+        Args:
+            samples: Number of samples to obtain
+            timeout: Timeout after which sampling should be stopped.
+                     If None: timemout[s] = samples * 1s + 5s
+                     If int: Timeout in [s]
+
+        Returns: None
+        """
+        if not timeout:
+            timeout = samples + 5.0
+
+        args = f"-O csv --samples {samples}".split(" ")
+        self._call_with_driver(*args)
+        self._timeout = Timeout(timeout)
+        self._running = True
+
+    @Driver.check_active
+    @step(result=True)
+    def get_samples(self):
+        """
+        Waits for sigrok to complete and returns all samples obtained afterwards.
+        This function blocks until either sigrok has terminated or the timeout has been reached.
+
+        Returns:
+            (unit_spec, [sample, ...])
+        """
+        if not self._running:
+            raise RuntimeError("Capture was not started before get_samples was called.")
+        while not self._timeout.expired:
+            if self._process.poll() is not None:
+                # process has finished. no need to wait for the timeout
+                break
+            time.sleep(0.1)
+        else:
+            # process did not finish in time
+            self.log.info("sigrok-cli did not finish in time, increase timeout?")
+            self._process.kill()
+
+        res = []
+        unit = ""
+        for line in self._process.stdout.readlines():
+            line = line.strip()
+            if b';' in line:
+                # discard header information
+                continue
+            if not unit:
+                # the first line after the header contains the unit information
+                unit = line.decode()
+            else:
+                # all other lines are actual values
+                res.append(float(line))
+        self._process.communicate()
+        return unit, res
```

Comment:
It's even worse. `start()` can be called again with `running==True` since it does not check that state. But calling `stop()` again without re-starting capture will probably result in an immediate timeout and return no samples.

So i will to both: set `running=False` in `stop()` and check for `running` in `start()`.